design-documents/graph-ql/coverage/company.md

```diff
@@ -36,7 +36,10 @@ type Company @doc(description: "Company entity output data schema.") {
     ): CompanyRoles!  @doc(description: "Returns the list of defined roles at Company.")
     role(id: ID!): CompanyRole  @doc(description: "Returns company role by id.")
     acl_resources: [CompanyAclResource]  @doc(description: "Returns the list of all permission resources.")
-    hierarchy: CompanyHierarchyOutput  @doc(description: "Returns the complete data about company structure.")
+    structure(
+        rootID: ID = 0 @doc(description: "Tree depth to begin query")
```

Comment:
`rootID` and `depth` together allow fetching results to any depth, starting at any level of the tree. Useful for lazy loading

Reply:
There was another suggestion from someone offline to support `rootIDs`instead of `rootID`, so you could get multiple subtrees in a single query.

Any input on that welcome. I don't _think_ it's necessary - can't think of a compelling use-case where you need 2 separate subtrees but not the parent node and other siblings.